### PR TITLE
Gyroscope

### PR DESCRIPTION
**Pull request by elizaBarnett**

### CODE REVIEW

File: src/main/java/frc/robot/commands/GyroTurnCommand.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        double currentAngle = gyro.getGyroAngle();
+        newAngle = currentAngle + turnAngle;
+        drive.setBrakeMode();
+    }
+
+    // Called repeatedly when this Command is scheduled to run
+    @Override
+    public void execute() {
+                double error = newAngle - gyro.getGyroAngle();
+                double speed = -error*(0.0027) *.65;
```
Comment (nbgray) [inline src/main/java/frc/robot/commands/GyroTurnCommand.java]: All these doubles should be assigned at the top of the class with a comment describing them

File: src/main/java/frc/robot/commands/GyroTurnCommand.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                if(Math.abs(speed) < .1){
```
Comment (nbgray) [inline src/main/java/frc/robot/commands/GyroTurnCommand.java]: Make a constant

File: src/main/java/frc/robot/subsystems/drive/Gyroscope.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        return imu.getAngle() - drift;
+    }
+
+    public void resetGyro(){
+        imu.reset();
+        startTime = Timer.getFPGATimestamp();
+    }
+
+    public void driveStraightGyro(double forwardSpeed,double desiredAngle){ 
+        double error = (desiredAngle-getGyroAngle()) * .0014;
```
Comment (nbgray) [inline src/main/java/frc/robot/subsystems/drive/Gyroscope.java]: magic number

File: src/main/java/frc/robot/subsystems/drive/Gyroscope.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    public void driveStraightGyro(double forwardSpeed,double desiredAngle){ 
+        double error = (desiredAngle-getGyroAngle()) * .0014; 
+        drive.arcadeDrive(forwardSpeed, error);
+    }
+
+    public void backToStartingAngle(double speed){
+        double currentAngle = getGyroAngle();
+        double turnAngle = currentAngle % 360;
```
Comment (nbgray) [inline src/main/java/frc/robot/subsystems/drive/Gyroscope.java]: doubles should be x.x to stop an implicit casting, ie 360.0

File: src/main/java/frc/robot/subsystems/drive/Gyroscope.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        if( turnAngle/ Math.abs(turnAngle) == 1){
```
Comment (nbgray) [inline src/main/java/frc/robot/subsystems/drive/Gyroscope.java]: it would be better to do 

if( turnAngle < 0.0)
{
speed = -1.0 * speed;
}

If your speed is ever 0 this will have a divide by 0 error.

File: src/main/java/frc/robot/subsystems/drive/Gyroscope.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+            speed = speed; //speed stays positive
+        }
+        else{
+        speed = -speed; //speed is negative
+        }
+        drive.setBrakeMode();
+        while(Math.abs(getGyroAngle()) < Math.abs(currentAngle)){
+            drive.arcadeDrive(-0, -speed); //b/c joysticks come out negative, we can either change the arcade method, or just make all things plugged in negative
```
Comment (nbgray) [inline src/main/java/frc/robot/subsystems/drive/Gyroscope.java]: -0.0 instead of -0

File: src/main/java/frc/robot/Robot.java [hunk 7]
```diff
@@ -9,20 +9,24 @@
...
   private static final String kDefaultAuto = "Default";
   private static final String kCustomAuto = "My Auto";
   private String m_autoSelected;
   private final SendableChooser<String> m_chooser = new SendableChooser<>();
 
   private Drive drive;
   private Controls controls;
+  private Gyroscope gyro;
   private AutonomousCommand autonomousCommand;
-  private TeleopCommand teleopCommand;
```
Comment (nbgray) [inline src/main/java/frc/robot/Robot.java]: If you remove teleopCommand how will the robot drive?

File: src/main/java/frc/robot/Robot.java [hunk 8]
```diff
@@ -9,20 +9,24 @@
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.CommandScheduler;
 import frc.robot.subsystems.drive.Drive;
+import frc.robot.subsystems.drive.Gyroscope;
 import frc.robot.commands.AutonomousCommand;
 import frc.robot.commands.TeleopCommand;
```
Comment (nbgray) [inline src/main/java/frc/robot/Robot.java]: If you're not using a class, you should remove the import

File: src/main/java/frc/robot/Robot.java [hunk 9]
```diff
@@ -101,10 +113,8 @@ public void autonomousPeriodic() {
 
   @Override
   public void teleopInit() {
-    this.teleopCommand = new TeleopCommand(controls, drive);
-    if (teleopCommand != null) {
-      teleopCommand.schedule();
-    }
+    gyro.resetGyro();
+   // drive.setDefaultCommand(new TeleopCommand(controls,drive, gyro));
```
Comment (nbgray) [inline src/main/java/frc/robot/Robot.java]: How does the robot drive if there's no teleop command?

File: src/main/java/frc/robot/commands/GyroTurnCommand.java (hunk 1 above)
Comment (elizaBarnett) [reply src/main/java/frc/robot/commands/GyroTurnCommand.java]: moved and commented

File: src/main/java/frc/robot/commands/GyroTurnCommand.java (hunk 2 above)
Comment (elizaBarnett) [reply src/main/java/frc/robot/commands/GyroTurnCommand.java]: Done

File: src/main/java/frc/robot/subsystems/drive/Gyroscope.java (hunk 4 above)
Comment (elizaBarnett) [reply src/main/java/frc/robot/subsystems/drive/Gyroscope.java]: Deleted back to starting angle method in gyro class, made it its own command, mapped it to the b button, and fixed the double formating

Comment (elizaBarnett) [reply src/main/java/frc/robot/subsystems/drive/Gyroscope.java]: in the two gyro commands

File: src/main/java/frc/robot/subsystems/drive/Gyroscope.java (hunk 6 above)
Comment (elizaBarnett) [reply src/main/java/frc/robot/subsystems/drive/Gyroscope.java]: done

File: src/main/java/frc/robot/Robot.java (hunk 9 above)
Comment (elizaBarnett) [reply src/main/java/frc/robot/Robot.java]: it glitches occassionally and spins and eventually straightens out again, the encoder drive straight is more reliable. Which means when we merge it I'll save that old code

File: src/main/java/frc/robot/subsystems/drive/Gyroscope.java (hunk 3 above)
Comment (elizaBarnett) [reply src/main/java/frc/robot/subsystems/drive/Gyroscope.java]: fixed

File: src/main/java/frc/robot/subsystems/drive/Gyroscope.java (hunk 5 above)
Comment (elizaBarnett) [reply src/main/java/frc/robot/subsystems/drive/Gyroscope.java]: I got ride of this in gyroscope, but I'm working on finishing this issue in the gyro commands

File: src/main/java/frc/robot/commands/ReorientToFieldCommand.java [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Override
+    public void initialize() {
+        drive.setBrakeMode();
+    }
+
+    // Called repeatedly when this Command is scheduled to run
+    @Override
+    public void execute() {
+                error = gyro.getGyroAngle() % 360.0;
```
Comment (nbgray) [inline src/main/java/frc/robot/commands/ReorientToFieldCommand.java]: Your code is good, but I think the gyro has way too much drift for this to actually work.  But I would be happy to be wrong!

File: src/main/java/frc/robot/subsystems/drive/Gyroscope.java [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+    public void driveStraightGyro(double forwardSpeed,double desiredAngle){ 
+        double error = (desiredAngle-getGyroAngle()) * pValue; 
+        drive.arcadeDrive(forwardSpeed, error);
+    }
+
+    public void gyroCalibrate(){
+        startTime = Timer.getFPGATimestamp();        
+        double startAngle = imu.getAngle();
+        try{
+            Thread.sleep(16000);
```
Comment (nbgray) [inline src/main/java/frc/robot/subsystems/drive/Gyroscope.java]: Make this configurable.  Maybe part of the constructor?

Comment (elizaBarnett) [reply src/main/java/frc/robot/subsystems/drive/Gyroscope.java]: do you want me to make the sleep configurable?

File: src/main/java/frc/robot/subsystems/drive/Drive.java [hunk 12]
```diff
@@ -3,20 +3,23 @@
...
   public Encoder leftEncoder;
   public Encoder rightEncoder;
 
   private Gearbox leftGearbox;
   private Gearbox rightGearbox;
   // left gear box CAN ids
-  private final int LEFT_BACK_CAN_ID = 6;
-  private final int LEFT_FRONT_CAN_ID = 12;
+  private final int LEFT_BACK_CAN_ID = 6;//6
+  private final int LEFT_FRONT_CAN_ID = 5;//12
```
Comment (nbgray) [inline src/main/java/frc/robot/subsystems/drive/Drive.java]: Left is now:

  private final int LEFT_BACK_CAN_ID = 11;
  private final int LEFT_FRONT_CAN_ID = 12;
  //  private final int LEFT_MIDDLE_CAN_ID = 13;

File: src/main/java/frc/robot/subsystems/drive/Drive.java [hunk 13]
```diff
@@ -3,20 +3,23 @@
...
   // right gear box CAN ids
-  private final int RIGHT_BACK_CAN_ID = 4;
-  private final int RIGHT_FRONT_CAN_ID = 5;
+  private final int RIGHT_BACK_CAN_ID = 8;//4
```
Comment (nbgray) [inline src/main/java/frc/robot/subsystems/drive/Drive.java]: Right is now:

  private final int RIGHT_BACK_CAN_ID = 4;
  private final int RIGHT_FRONT_CAN_ID = 5;
  //  private final int RIGHT_MIDDLE_CAN_ID = 6;

**Comment by kushagharahi** [comment]
Can you resolve the conflicts?

File: src/main/java/frc/robot/Robot.java [hunk 14]
```diff
@@ -31,6 +35,8 @@
   private double maxSpeed;
   @Log
   private double distance;
+
+  boolean drivingStraight = false;
```
Comment (kushagharahi) [inline src/main/java/frc/robot/Robot.java]: is this used anywhere?

File: src/main/java/frc/robot/Robot.java [hunk 15]
```diff
@@ -42,9 +48,10 @@ public void robotInit() {
     SmartDashboard.putData("Auto choices", m_chooser);
 
     Logger.configureLoggingAndConfig(this, false);
-
     this.drive = new Drive();
     this.controls = new Controls(new Joystick(JOYSTICK_PORT));
+    this.gyro = new Gyroscope(drive);
+    this.io = new IO(controls,drive, gyro);
```
Comment (kushagharahi) [inline src/main/java/frc/robot/Robot.java]:
```suggestion
    this.io = new IO(controls, drive, gyro);
```

File: src/main/java/frc/robot/Robot.java (hunk 9 above)
Comment (kushagharahi) [reply src/main/java/frc/robot/Robot.java]: I don't think this is how merging will work. If you want the old encoder code you will need to make sure you're not touching that code. Since this branch is ahead of master, the changes in here will take president

File: src/main/java/frc/robot/commands/TeleopCommand.java [hunk 16]
```diff
@@ -3,38 +3,52 @@
...
     private Drive drive;
+    private Gyroscope gyroscope;
     private Controls controls;
     private double DEADZONE = 0.11;
-    private boolean resetEncoders;
+    double desiredAngle;
+    boolean drivingStraight = false;
   
-    public TeleopCommand(Controls controls, Drive drive) {
+    public TeleopCommand(Controls controls, Drive drive, Gyroscope gyroscope) {
```
Comment (kushagharahi) [inline src/main/java/frc/robot/commands/TeleopCommand.java]: If this drive straight code doesn't work I suggest you revert these changes so you don't overwrite the changes in master.

File: src/main/java/frc/robot/subsystems/drive/Gyroscope.java [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+
+public class Gyroscope extends SubsystemBase {
+    //Classes
+    private Drive drive; 
+    private ADIS16470_IMU imu;
+    //variables
+    private double startTime;
+    private double driftPerSecond;
+    //constants
+    private double pValue = .0014;
```
Comment (kushagharahi) [inline src/main/java/frc/robot/subsystems/drive/Gyroscope.java]:
```suggestion
    private final double P_VALUE = .0014;
```

File: src/main/java/frc/robot/Robot.java [hunk 18]
```diff
@@ -91,34 +100,21 @@ public void autonomousInit() {
...
   public void autonomousPeriodic() {
     CommandScheduler.getInstance().run();
   }
 
   @Override
   public void teleopInit() {
-    this.teleopCommand = new TeleopCommand(controls, drive);
-    if (teleopCommand != null) {
-      teleopCommand.schedule();
-    }
```
Comment (kushagharahi) [inline src/main/java/frc/robot/Robot.java:113]: This comment: https://github.com/PearceRobotics/infinite-recharge-2020/pull/54/files#r379175534 applies here as well.

File: src/main/java/frc/robot/commands/GyroTurnCommand.java [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+                // do nothing
+            }
+        }
+        drive.arcadeDrive(0.0, speed);
+    }
+
+    // Make this return true when this Command no longer needs to run execute()
+    @Override
+    public boolean isFinished() {
+        if(Math.abs(gyro.getGyroAngle() - newAngle) > 1.0)
```
Comment (kushagharahi) [inline src/main/java/frc/robot/commands/GyroTurnCommand.java]: I suggest making this 1.0 deadband a final variable at the top of this class.

File: src/main/java/frc/robot/commands/TeleopCommand.java [hunk 20]
```diff
@@ -3,38 +3,52 @@
...
-        System.out.println("Initialized");
-
+        gyroscope.resetGyro();
     }
 
     // Called repeatedly when this Command is scheduled to run
     @Override
     public void execute() {
+        System.out.println(gyroscope.getGyroAngle());
         if (controls.getRightX(DEADZONE) == 0.0) {
```
Comment (kushagharahi) [inline src/main/java/frc/robot/commands/TeleopCommand.java]: Having if statements to see what control button you're pressing defeats the purpose of command based programming. In command based programming we want to use the `whenPressed()` function in your IO class in order to let the system schedule the commands. Sorry that we didn't catch this when you merged your encoder code.

The appropriate way to handle this is check your turn modifier value in your arcade drive method and apply your drive straight code if it's equal to 0. It's the same idea except you're not checking your controls. This command should not be aware of the Controls class.

Comment (kushagharahi) [reply src/main/java/frc/robot/commands/TeleopCommand.java]: This might be the reason your drive straight code doesn't work and makes it spin.

File: src/main/java/frc/robot/subsystems/drive/Gyroscope.java [hunk 21]
```diff
@@ -0,0 +1,53 @@
...
+        double drift = runTime * driftPerSecond;
+        return imu.getAngle() - drift;
+    }
+
+    public void resetGyro(){
+        imu.reset();
+        startTime = Timer.getFPGATimestamp();
+    }
+
+    public void driveStraightGyro(double forwardSpeed,double desiredAngle){
```
Comment (kushagharahi) [inline src/main/java/frc/robot/subsystems/drive/Gyroscope.java]: instead of your gyro class being aware of the drive class, can the drive class utilize the gyro class instead?

**Review by nbgray — APPROVED** [review]

**Review by nbgray — APPROVED** [review]

File: src/main/java/frc/robot/subsystems/drive/Drive.java [hunk 22]
```diff
@@ -84,4 +119,8 @@ public double straightTurnPower(){
     double turnPower = error * .15;
     return turnPower;
   }
+
+  public double driveStraightGyro(double desiredAngle){
```
Comment (kushagharahi) [inline src/main/java/frc/robot/subsystems/drive/Drive.java]: Let’s give method a better name. Maybe something like getAngularDelta or getAngularDrift?

File: src/main/java/frc/robot/subsystems/drive/Drive.java [hunk 23]
```diff
@@ -84,4 +119,8 @@ public double straightTurnPower(){
     double turnPower = error * .15;
     return turnPower;
   }
+
+  public double driveStraightGyro(double desiredAngle){  
+    return  (desiredAngle - gyroscope.getGyroAngle()) * P_VALUE;
```
Comment (kushagharahi) [inline src/main/java/frc/robot/subsystems/drive/Drive.java]: Let’s just move this line to the drive straight part of arcade drive instead of having it in its own method

File: src/main/java/frc/robot/subsystems/drive/Drive.java [hunk 24]
```diff
@@ -3,24 +3,38 @@
...
+  private Encoder leftEncoder;
+  private Encoder rightEncoder;
   private Gearbox leftGearbox;
   private Gearbox rightGearbox;
+
+  private Gyroscope gyroscope;
+
+  //variables
+  private double desiredAngle;
+  private double turnModifer;
```
Comment (kushagharahi) [inline src/main/java/frc/robot/subsystems/drive/Drive.java]: You don’t need this variable

File: src/main/java/frc/robot/subsystems/drive/Drive.java [hunk 25]
```diff
@@ -60,13 +83,25 @@ public void resetEncoders() {
...
+    brakeMode = true;
+    }
+    else{
+      //do nothing
+    }
   }
 
   public void setCoastMode(){
+    if(brakeMode == true){
     leftGearbox.setCoastMode();
```
Comment (kushagharahi) [inline src/main/java/frc/robot/subsystems/drive/Drive.java]: This if statements inner brackets need some tab love

File: src/main/java/frc/robot/subsystems/drive/Drive.java (hunk 22 above)
Comment (kushagharahi) [inline src/main/java/frc/robot/subsystems/drive/Drive.java]: Give this a better name. Maybe something like getAngularDriftError or getAngularDeltaError?

File: src/main/java/frc/robot/subsystems/drive/Drive.java [hunk 26]
```diff
@@ -60,13 +83,25 @@ public void resetEncoders() {
   }
 
   public void setBrakeMode(){
+    if(brakeMode == false){
     leftGearbox.setBrakeMode();
```
Comment (kushagharahi) [inline src/main/java/frc/robot/subsystems/drive/Drive.java:86]: This if statement also needs tab love

File: src/main/java/frc/robot/subsystems/drive/Gearbox.java [hunk 27]
```diff
@@ -1,11 +1,11 @@
-
 package frc.robot.subsystems.drive;
 
 import com.revrobotics.CANSparkMax;
 import com.revrobotics.CANError;
 import com.revrobotics.CANSparkMax.IdleMode;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
-public class Gearbox {
+public class Gearbox extends SubsystemBase{
```
Comment (kushagharahi) [inline src/main/java/frc/robot/subsystems/drive/Gearbox.java]: Since we don’t use the gearboxes directly in a command, does it need to extend SubsystemBase?

File: src/main/java/frc/robot/subsystems/drive/Drive.java [hunk 28]
```diff
@@ -60,13 +83,25 @@ public void resetEncoders() {
...
     rightGearbox.setBrakeMode();
+    brakeMode = true;
+    }
+    else{
```
Comment (kushagharahi) [inline src/main/java/frc/robot/subsystems/drive/Drive.java]: This empty else is not needed

File: src/main/java/frc/robot/Robot.java [hunk 29]
```diff
@@ -77,6 +84,7 @@ public void robotPeriodic() {
   @Override
   public void autonomousInit() {
     this.autonomousCommand = new AutonomousCommand(distance, maxSpeed, drive);
+    gyro.resetGyro();
```
Comment (kushagharahi) [inline src/main/java/frc/robot/Robot.java]: This isnt needed

File: src/main/java/frc/robot/commands/GyroTurnCommand.java [hunk 30]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    // Called repeatedly when this Command is scheduled to run
+    @Override
+    public void execute() {
+
+        error = newAngle - gyro.getGyroAngle();
+        speed = -error * P_VALUE;
+
+        /*
```
Comment (kushagharahi) [inline src/main/java/frc/robot/commands/GyroTurnCommand.java]: You were right. Let’s keep this if statement.

File: src/main/java/frc/robot/commands/ReorientToFieldCommand.java [hunk 31]
```diff
@@ -0,0 +1,67 @@
...
+                speed = error*(ratio) * p;
+                if(Math.abs(speed) < minSpeed){
+                    if (speed < 0.0){
+                        speed = -minSpeed;
+                    }
+                    else if(speed > 0.0) {
+                        speed = minSpeed;
+                    }
+                    else{
```
Comment (kushagharahi) [inline src/main/java/frc/robot/commands/ReorientToFieldCommand.java]: This empty else isn’t needed

File: src/main/java/frc/robot/commands/ReorientToFieldCommand.java [hunk 32]
```diff
@@ -0,0 +1,67 @@
...
+                    }
+                }
+                drive.arcadeDrive(0.0, speed);
+        }
+
+    // Make this return true when this Command no longer needs to run execute()
+    @Override
+    public boolean isFinished() {
+        if((Math.abs(gyro.getGyroAngle())% 360.0) > 1.0)
+        {
```
Comment (kushagharahi) [inline src/main/java/frc/robot/commands/ReorientToFieldCommand.java]: Let’s stay with one bracket style

File: src/main/java/frc/robot/subsystems/drive/Drive.java [hunk 33]
```diff
@@ -60,13 +83,25 @@ public void resetEncoders() {
...
+      //do nothing
+    }
   }
 
   public void setCoastMode(){
+    if(brakeMode == true){
```
Comment (kushagharahi) [inline src/main/java/frc/robot/subsystems/drive/Drive.java]: You don’t actually need to know what mode the gearboxes are. You can just blindly set them to coast and brake mode

File: src/main/java/frc/robot/subsystems/drive/Drive.java [hunk 34]
```diff
@@ -60,13 +83,25 @@ public void resetEncoders() {
   }
 
   public void setBrakeMode(){
+    if(brakeMode == false){
```
Comment (kushagharahi) [inline src/main/java/frc/robot/subsystems/drive/Drive.java]: You don’t actually need to know what mode the gearboxes are. You can just blindly set them to coast and brake mode

File: src/main/java/frc/robot/commands/ReorientToFieldCommand.java [hunk 35]
```diff
@@ -0,0 +1,67 @@
...
+
+public class ReorientToFieldCommand extends CommandBase{
+    //Classes
+    private Drive drive;
+    private Gyroscope gyro;
+    //Variables
+    private double error; //error the robot is off its original position (from 0 to 360)
+    private double speed;//speed the robot turns
+    //Constants
+    private double ratio = 0.0027;//ratio to multiply error by to get a number between -1 and 1 for the speed
```
Comment (kushagharahi) [inline src/main/java/frc/robot/commands/ReorientToFieldCommand.java]: Let’s make the same p value changes we made in the gyroturncommand

File: src/main/java/frc/robot/commands/ReorientToFieldCommand.java [hunk 36]
```diff
@@ -0,0 +1,67 @@
...
+import frc.robot.subsystems.drive.Drive;
+import frc.robot.subsystems.drive.Gyroscope;
+
+public class ReorientToFieldCommand extends CommandBase{
+    //Classes
+    private Drive drive;
+    private Gyroscope gyro;
+    //Variables
+    private double error; //error the robot is off its original position (from 0 to 360)
+    private double speed;//speed the robot turns
```
Comment (kushagharahi) [inline src/main/java/frc/robot/commands/ReorientToFieldCommand.java]: This doesn’t need to be a class variabel

File: src/main/java/frc/robot/commands/ReorientToFieldCommand.java [hunk 37]
```diff
@@ -0,0 +1,67 @@
...
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.drive.Drive;
+import frc.robot.subsystems.drive.Gyroscope;
+
+public class ReorientToFieldCommand extends CommandBase{
+    //Classes
+    private Drive drive;
+    private Gyroscope gyro;
+    //Variables
+    private double error; //error the robot is off its original position (from 0 to 360)
```
Comment (kushagharahi) [inline src/main/java/frc/robot/commands/ReorientToFieldCommand.java]: This doesn’t need to be a class variable

File: src/main/java/frc/robot/commands/ReorientToFieldCommand.java [hunk 38]
```diff
@@ -0,0 +1,67 @@
...
+                        // do nothing
+                    }
+                }
+                drive.arcadeDrive(0.0, speed);
+        }
+
+    // Make this return true when this Command no longer needs to run execute()
+    @Override
+    public boolean isFinished() {
+        if((Math.abs(gyro.getGyroAngle())% 360.0) > 1.0)
```
Comment (kushagharahi) [inline src/main/java/frc/robot/commands/ReorientToFieldCommand.java]: Let’s make this 1.0 deadband a final class variable

File: src/main/java/frc/robot/commands/GyroTurnCommand.java [hunk 39]
```diff
@@ -0,0 +1,67 @@
...
+
+public class GyroTurnCommand extends CommandBase {
+    // Classes
+    private final Gyroscope gyro;
+    private final Drive drive;
+    // Variables
+    private final double turnAngle;// Angle we want the robot to rotate in degrees
+    private double newAngle;// Angle gyroscope will see new angle as
+    private double error; // continuous error the robot is off from its desired position
+    private double speed;// speed the robot turns
```
Comment (kushagharahi) [inline src/main/java/frc/robot/commands/GyroTurnCommand.java]: These 4 class variables arent needed

File: src/main/java/frc/robot/commands/GyroTurnCommand.java [hunk 40]
```diff
@@ -0,0 +1,67 @@
...
+    // Constants
```
Comment (kushagharahi) [inline src/main/java/frc/robot/commands/GyroTurnCommand.java:14]: Speed and error aren’t needed as class variables.

**Review by kushagharahi — APPROVED** [review]
